### ISSUE TITLE
Future @cb_jarvis upgrades

### ISSUE THREAD

**Issue by lpatmo** [codebuddies/codebuddies#958]
(Thanks @julianjohannesen for feedback)

- pass in description
- pass in hangout type (silent, collab, teaching)
- pass in external link in case folks want to use Zoom, LiveStream, etc.

**Comment by gauravchl** [comment]
More command ideas for jarvis:

1. `@cb_jarvis stats`
    o/p: Total hangouts organized, learnings reported, study groups, discussions, online users, total users etc...
2. `@cb_jarvis 24/7 hangouts`
    o/p: all 24/7 hangout detail(title/studyGroupName) with jitsi link
3. `@cb_jarvis #TodayILearned`
   o/p: Random "today i learned" post from last 10 posts. With user name.

**Comment by lpatmo** [comment]
Love all those ideas. If they can be pulled into separate PRs/issues, we can definitely do that so that you (and potentially other contributors!) get max credit. :)

**Comment by gauravchl** [comment]
Good idea. So let's create new issues for every command with `#help wanted` and `#good first issue`?
https://help.github.com/articles/helping-new-contributors-find-your-project-with-labels/

Does [theses projects](https://github.com/codebuddies/codebuddies/projects) are still relevant? 
May be we could keep above tasks well organized there as well.

**Comment by lpatmo** [comment]
> Good idea. So let's create new issues for every command with #help wanted and #good first issue?

Sounds good! I will start doing that. 

> Does theses projects are still relevant?

Er... wow, I totally forgot about those projects. Thanks for reminding me. I think some of those issues are relevant still, but working with projects hasn't been very intuitive, so I will try to turn them into issues. I've mostly still been using the Trello board at https://trello.com/b/3Olzq5sh/codebuddies.

That said, if you like the Github Projects feature, I don't trying it out! I'll create a new board called `@cb_jarvis: Slack to CodeBuddies` or something.

**Comment by railsstudent** [comment]
@lpatmo @gauravchl  Should create hangout command split into create silent hangout, create collab hangout and create teaching hangout commands respectively or add a new segment to create hangout to hold hangout type?

**Comment by lpatmo** [comment]
Sadly, not many people are using the `@cb_jarvis` command to create new hangouts. Besides `hangout type`, there are two other arguments we could add:
- specify the description
- specify an optional external link (like Twitch or Zoom) - requested by @julianjohannesen 

I'm not sure which of all of the above would be most important, though, if our ultimate goal was to encourage more usage of `cb_jarvis`. Do you have a preference (or thoughts)? 

I worry about you or others spending time extending this with not much usage, but if the end goal is to make the PR for the sake of the PR (or simply because it would be fun to add), that's understandable. :) 

Maybe `cb_jarvis` usage simply needs to be adopted better into the culture of the community by some active users. In the meantime, my prediction is `@cb_jarvis til message` will see a lot more usage since it's a lower-expectation ask from the start.

**Comment by railsstudent** [comment]
@lpatmo I believe optional external link has highest priority because Julian requested it and he is an active hangout creator.

I believe we have to advertise cb_jarvis in slack channels such that people can develop habit of using slack to create hangout. 
It depends on their willingness to  context switch between slack and codebuddies.org

**Comment by lpatmo** [comment]
> I believe optional external link has highest priority because Julian requested it and he is an active hangout creator.

Yeah, makes sense :)

**Comment by gauravchl** [comment]
@lpatmo @railsstudent Personally i'm also finding it difficult to mention cb_jarvis in slack and perform any action. 
I think it would be way much easier to just perform actions directly with hash tags.
Like if sentence starts with `#createHangout` or `#listHangout` or `#til`

So i'm thinking to update the existing solution a bit and add support for above solutions. 

Let me know if what do you think.

**Comment by gauravchl** [comment]
```
#createHangout today 9am, advent of code
#til How to run onion inside docker
```

**Comment by lpatmo** [comment]
Agree! Maybe a refactor can happen after https://github.com/codebuddies/codebuddies/pull/1083 is merged?

**Comment by gauravchl** [comment]
@lpatmo Yeah, of course.

**Comment by gauravchl** [comment]
@lpatmo Sweet! I will start working on hashtag version.

**Comment by lpatmo** [comment]
Thank you! I am due to pick a time to work on https://github.com/sourcecred/sourcecred/issues/576 (wrapping up inboxes); perhaps all of this can go out in the same release within the next couple of weeks. :)

**Comment by railsstudent** [comment]
It could be the start of partnership between both projects;  codebuddies contributes to the project and come back to share in hangout.

**Comment by lpatmo** [comment]
@railsstudent Can you elaborate on what you mean?

**Comment by railsstudent** [comment]
@lpatmo  People in codebuddies help fix issues in sourcecred project and share their experiences in hangout to attract interest from other members to help contribute. If number of contributions rise through this method, other maintainers will come to codebuddies to have code walkthrough and look for volunteers.
